RUFAS/routines/EEE/emissions.py

```diff
@@ -475,32 +468,34 @@ def _calculate_emissions_by_field(
         self,
         feeds_grown: list[dict[str, Any]],
         field_emissions: dict[str, float],
-        fertilizer_applications: dict[str, float],
         manure_applications: dict[str, float],
         manure_requests: dict[str, float],
+        fertilizer_applications_data: list[dict[str, Any]],
     ) -> list[dict[str, Any]]:
         """
-        Partitions emissions from the field where crops/feeds were grown to those crops based on their relative mass.
+        Partitions emissions from the field where crops/feeds were grown to those crops.
         """
         field_size = feeds_grown[0]["field_size"]
         total_dry_mass_per_ha_grown = sum([crop["dry_yield"] for crop in feeds_grown])
 
+        for crop in feeds_grown:
+            crop["nitrous_oxide_emissions"] = 0.0
+            crop["ammonia_emissions"] = 0.0
+            crop["carbon_stock_change"] = 0.0
+            crop["nitrogen_fertilizer_used"] = 0.0
+            crop["nitrogen_fertilizer_embedded_CO2_emissions"] = 0.0
+            crop["phosphorus_fertilizer_used"] = 0.0
+            crop["phosphorus_fertilizer_embedded_CO2_emissions"] = 0.0
+            crop["potassium_fertilizer_used"] = 0.0
+            crop["potassium_fertilizer_embedded_CO2_emissions"] = 0.0
+            crop["manure_nitrogen_used"] = 0.0
+            crop["manure_nitrogen_requested"] = 0.0
+
         if total_dry_mass_per_ha_grown == 0.0:
-            for crop in feeds_grown:
-                crop["nitrous_oxide_emissions"] = 0.0
-                crop["ammonia_emissions"] = 0.0
-                crop["carbon_stock_change"] = 0.0
-                crop["nitrogen_fertilizer_used"] = 0.0
-                crop["nitrogen_fertilizer_embedded_CO2_emissions"] = 0.0
-                crop["phosphorus_fertilizer_used"] = 0.0
-                crop["phosphorus_fertilizer_embedded_CO2_emissions"] = 0.0
-                crop["potassium_fertilizer_used"] = 0.0
-                crop["potassium_fertilizer_embedded_CO2_emissions"] = 0.0
-                crop["manure_nitrogen_used"] = 0.0
-                crop["manure_nitrogen_requested"] = 0.0
             return feeds_grown
 
-        for crop in feeds_grown:
+        sorted_crops = sorted(feeds_grown, key=lambda crop: (crop["planting_year"], crop["planting_day"]))
```

Comment:
I recommend converting planting year and planting day to a datetime.date instance and have those be used to sort things. I'm not sure that using tuples here wouldn't work but using date instances makes it much more robust.
```suggestion
        sorted_crops = sorted(feeds_grown, key=lambda crop: (crop["planting_year"], crop["planting_day"]))
```

Reply:
Can you expand a little on what converting these to datetime.date objects would do to make this more robust and in what cases we might want to have this robustness available here? I don't think it would be too difficult to do but it seems like an added step/overhead of object creation for a list-sort that's working as is.

Reply:
To me, sorting a list of tuples is not as intuitive as sorting a list of dates. So if there is a strange edge case that would mess with the sorting routine, I expect it would happen with something less intuitive like tuples than it would with dates, especially because dates have a library dedicated to making sure they can captured and used correctly.

This was my gut reaction to reading this code, in part because I introduced a serious bug into a different part of this module by trying to compare dates more directly with numbers than with date objects. I looked up how Python sorts tuples just now and I think your method will be bug free.

> I don't think it would be too difficult to do but it seems like an added step/overhead of object creation for a list-sort that's working as is.

This is a very good point, I think either way would be fine but sticking with the tuples would have that edge of being faster.